### ISSUE TITLE
 u'[script] unknown field [file], parser not found' 

### ISSUE THREAD

**Issue by SHSauler** [google/timesketch#585]
* Ubuntu 16.04
* timesketch 20170721
* TL/DR: `Error` on GUI when commenting

Hi guys,

When we try to comment on log lines, we experience `Error`s in timesketch on the GUI with the following in flask:

````python
RequestError: TransportError(400, u'illegal_argument_exception', u'[script] unknown field [file], parser not found')
````

What we tried on a new & updated Ubuntu 16.10 each time, following your Vagrant build script and the [Skadi script](https://github.com/orlikoski/Skadi/blob/master/scripts/signedbuildskadi.sh) closely:

* First try: Installed timesketch via `pip`
* Second try: Installed timesketch via `pip`, then updated via repo
* Third try: Installed via repo in the first place
* Fourth try: The vagrant deploy script. Did not go through.

Always leads to the same issue.

Database issue? Trying to update the database via `tsctl db update` as per documentation leads to:

```python
qlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) column "group_id" of relation "searchindex_accesscontrolentry" already exists
 [SQL: 'ALTER TABLE searchindex_accesscontrolentry ADD COLUMN group_id INTEGER']
```

Thanks and kind regards!

**Comment by mari0d** [comment]
What version of Elasticsearch are you using?

**Comment by SHSauler** [comment]
```
elasticsearch:all/stable 6.2.4 uptodate
jq:amd64/xenial 1.5+dfsg-1 uptodate
jq:i386 not installed
libffi-dev:amd64/xenial 3.2.1-4 uptodate
libffi-dev:i386 not installed
neo4j:all/stable 1:3.3.5 uptodate
openjdk-8-jre-headless:amd64/xenial-security 8u162-b12-0ubuntu0.16.04.2 uptodate
openjdk-8-jre-headless:i386 not installed
plaso-tools:all/xenial 20180127-1ppa1~xenial uptodate
postgresql:all/xenial-security 9.5+173ubuntu0.1 uptodate
python-dev:amd64/xenial-updates 2.7.12-1~16.04 uptodate
python-dev:i386 not installed
python-pip:all/xenial-updates 8.1.1-2ubuntu0.4 uptodate
python-plaso:all/xenial 20180127-1ppa1~xenial uptodate
python-psycopg2:amd64/xenial 2.6.1-1build2 uptodate
python-psycopg2:i386 not installed
redis-server:amd64/xenial 4:4.0.9-1chl1~xenial1 uptodate
redis-server:i386 not installed
```

Versions via `pip list` match the requirements.txt

**Comment by mari0d** [comment]
Ok, yah, TimeSketch doesn't officially support ES 6.x currently I believe. The error you are getting is due to breaking changes in how ES 6.x handles scripting. Take a look at https://github.com/google/timesketch/issues/575 for more info.

**Comment by adamjnichols** [comment]
HI there -- are you using your own ES install with the vagrant run?

**Comment by SHSauler** [comment]
Closing this, since we are using ES 6.x. Thanks for your advice @mari0d and @adamjnichols!

**Comment by berggren** [comment]
@SHSauler FYI: We already support ES 6.x with new painless scripts. They are available here:
https://github.com/google/timesketch/blob/master/contrib/add_label.painless

I'm making it a priority to have ES 6.x ready by default.

**Comment by SHSauler** [comment]
@berggren Thanks for your info :)

Could you please give us a hint how to deploy on ES 6.x with the painless scripts? 

* We run a new development version of `timesketch`
* We experimented putting the `*.painless` into `/etc/elasticsearch/scripts` with the groovy-stuff

Error persists:

```
RequestError: TransportError(400, u'illegal_argument_exception', u'[script] unknown field [file], parser not found')
```

**Comment by SHSauler** [comment]
We could get it to work by using https://github.com/ptitdoc/timesketch/tree/scriptfix_201805. Probably not a good idea, but it works 🤷‍♂️

**Comment by berggren** [comment]
Hey, yeah that will work. I'm working on a solution to this, so we can support both versions side by side.
